grakn-test/src/test/java/ai/grakn/test/engine/controller/GraknExecuteQueryControllerTest.java

```diff
@@ -0,0 +1,93 @@
+package ai.grakn.test.engine.controller;
+
+import static ai.grakn.util.REST.Request.KEYSPACE;
+import static ai.grakn.util.REST.Request.Graql.INFER;
+import static ai.grakn.util.REST.Request.Graql.LIMIT_EMBEDDED;
+import static ai.grakn.util.REST.Request.Graql.MATERIALISE;
+import static ai.grakn.util.REST.Request.Graql.QUERY;
+import static ai.grakn.util.REST.Response.ContentType.APPLICATION_JSON_GRAQL;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.response.Response;
+
+import ai.grakn.engine.GraknEngineConfig;
+import ai.grakn.engine.controller.GraqlController;
+import ai.grakn.engine.factory.EngineGraknGraphFactory;
+import ai.grakn.test.GraphContext;
+import ai.grakn.test.SparkContext;
+import ai.grakn.test.graphs.MovieGraph;
+import ai.grakn.util.REST;
+
+public class GraknExecuteQueryControllerTest {
+    
+    private Response sendQuery(String query) {
+        return sendQuery(query, APPLICATION_JSON_GRAQL, true, false, -1);
+    }
+    
+    private Response sendQuery(String query, 
+                               String acceptType, 
+                               boolean reasonser,
+                               boolean materialise, 
+                               int limitEmbedded) {
+        return RestAssured.with()
+            .queryParam(KEYSPACE, graphContext.graph().getKeyspace())
+            .queryParam(QUERY, query)
+            .queryParam(INFER, reasonser)
+            .queryParam(MATERIALISE, materialise)
+            .queryParam(LIMIT_EMBEDDED, limitEmbedded)
+            .accept(acceptType)
+            .post(REST.WebPath.Graph.ANY_GRAQL);
+    }
+    
+    private static EngineGraknGraphFactory engineGraknGraphFactory = 
+            EngineGraknGraphFactory.create(GraknEngineConfig.create().getProperties());
+    
+    @ClassRule
+    public static GraphContext graphContext = GraphContext.preLoad(MovieGraph.get());
+
+    @ClassRule
+    public static SparkContext sparkContext = SparkContext.withControllers(spark -> {
+        new GraqlController(engineGraknGraphFactory, spark);
+    });
+        
+    @Test
+    public void testMatchQuery() {
```

Comment:
I had a compute query but it has problems starting spark (nothing to do with engine or the controller or the test) so I took it out for the moment because I know Sheldon is getting rid of that dependency. Otherwise, yes, good idea to add bad requests (it returns 400). However, I'm not sure when the other controllers are going away. We discussed replacing them with GraphQL, which I spent some time investigating so we should meet and chat about it....